test/e2e/config_test.go

```diff
@@ -727,6 +731,40 @@ func TestUserWorkloadMonitorThanosRulerConfig(t *testing.T) {
 	}
 }
 
+// checkMonitorConsolePluginReachable makes sure that one of the pods at least can serve /plugin-manifest.json
+// TODO: This is better done via a readiness probe in the Deployment.
+func checkMonitorConsolePluginReachable(t *testing.T) {
+	err := framework.Poll(time.Second, 5*time.Minute, func() error {
+		host, cleanUp, err := f.ForwardPort(t, f.Ns, "monitoring-plugin", 9443)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer cleanUp()
+
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		}
+		resp, err := client.Get(fmt.Sprintf("https://%s/plugin-manifest.json", host))
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("fail to read response body: %w", err)
+		}
+
+		if resp.StatusCode != http.StatusOK {
```

Comment:
(nit) I would check the status code first and then verify that the response body length is > 0 (or even valid JSON?).